src/sql/workbench/api/browser/mainThreadModelViewDialog.ts

```diff
@@ -88,11 +93,18 @@ export class MainThreadModelViewDialog implements MainThreadModelViewDialogShape
 	public $setDialogDetails(handle: number, details: IModelViewDialogDetails): Thenable<void> {
 		let dialog = this._dialogs.get(handle);
 		if (!dialog) {
-			dialog = new Dialog(details.title, details.width);
-			let okButton = this.getButton(details.okButton);
-			let cancelButton = this.getButton(details.cancelButton);
-			dialog.okButton = okButton;
-			dialog.cancelButton = cancelButton;
+			dialog = new Dialog(details.title, details.width, details.dialogStyle, details.dialogPosition, details.renderHeader, details.renderFooter, details.dialogProperties);
+
+			/**
+			 * Only peform actions on footer if it is shown.
+			 */
+			if (details.renderFooter === true || details.renderFooter === undefined) {
+				let okButton = this.getButton(details.okButton);
+				let cancelButton = this.getButton(details.cancelButton);
```

Comment:
`
dialog.okButton = this.getButton(details.okButton);

dialog.cancelButton = this.getButton(details.cancelButton);
`